src/EditorFeatures/Core/RenameTracking/RenameTrackingTaggerProvider.TrackingSession.cs

```diff
@@ -76,22 +76,22 @@ public TrackingSession(
 
                 var asyncToken = _asyncListener.BeginAsyncOperation(GetType().Name + ".UpdateTrackingSessionAfterIsRenamableIdentifierTask");
 
-                _isRenamableIdentifierTask.SafeContinueWithFromAsync(
-                    async t =>
-                    {
-                        await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, _cancellationToken).NoThrowAwaitable();
+                SwitchToMainThreadAndUpdateTrackingSessionAsync(_isRenamableIdentifierTask).CompletesAsyncOperation(asyncToken);
+                QueueUpdateToStateMachine(stateMachine, _isRenamableIdentifierTask);
 
-                        // Avoid throwing an exception in this common case
-                        if (_cancellationToken.IsCancellationRequested)
-                            return;
+                return;
 
-                        stateMachine.UpdateTrackingSessionIfRenamable();
-                    },
-                    _cancellationToken,
-                    TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously,
```

Comment:
the prior logic was 'OnlyOnRanToCompletion' which is what 'await' does anyways.  So the conversion is sound.